### PR TITLE
Fixes #3277 - using assembly

### LINKED ISSUE #3277: about_using doesn't mention "using assembly" syntax
The about_using help topic doesn't mention the "using assembly &lt;assembly name&gt;" syntax.

Is that syntax not supported for PowerShell 6?

If it is, where would PowerShell search for and load assemblies from?

If it is not, then the documentation should note that the "using assembly" syntax is not supported in PowerShell 6, although it is supported in PowerShell 5 and PowerShell 5.1.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8ee0c5b0-8465-adcc-be1b-9e6aafa4199d
* Version Independent ID: bc27ba48-5641-d363-7fc4-06ea01efa9e2
* Content: [about_Using](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_using?view=powershell-6)
* Content Source: [reference/6/Microsoft.PowerShell.Core/About/About_Using.md](https://github.com/PowerShell/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/About_Using.md)
* Product: **powershell**
* GitHub Login: @joeyaiello
* Microsoft Alias: **jaiello**

### PR DESCRIPTION
**Pull request by sdwheeler**
# PR Summary

Fixes #3277 - using assembly
Fixes [AB#1670360](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1670360)

## PR Context

Select the type(s) of documents being changed.

**Cmdlet reference & about_ topics**
- [x] Version 7.x preview content
- [x] Version 7.0 content
- [x] Version 6 content
- [x] Version 5.1 content

**Conceptual articles**
- [ ] Fundamental conceptual articles
- [ ] Script sample articles
- [ ] DSC articles
- [ ] Gallery articles
- [ ] JEA articles
- [ ] WMF articles
- [ ] SDK articles

## PR Checklist

- [x] I have read the [contributors guide](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/CONTRIBUTING.md) and followed the style and process guidelines
- [x] PR has a meaningful title
- [x] PR is targeted at the _staging_ branch
- [x] All relevant versions updated
- [x] Includes content related to issues and PRs - see [Closing issues using keywords](https://help.github.com/en/articles/closing-issues-using-keywords).
- [x] This PR is ready to merge and is not **Work in Progress**
  - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the
    title and remove the prefix when the PR is ready.

### CODE REVIEW

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md [hunk 1]
```diff
@@ -1,46 +1,63 @@
...
+Allows you to indicate which namespaces are used in the session.
 
 ## LONG DESCRIPTION
 
 The `using` statement allows you to specify which namespaces are used in the
 session. Adding namespaces simplifies usage of .NET classes and member and
-allows you to import classes from modules.
+allows you to import classes from script modules and assemblies.
 
 The `using` statement needs to be the first statement in the script.
```
Comment (rjmholt) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md]: I'd change this to something like:

> `using` statements must come before any other statements in a script.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md [hunk 2]
```diff
@@ -1,46 +1,63 @@
...
 
-### Syntax
+The `using` statement should not be confused with the `using:` scope modifier
+for variables. For more information, see
+[about_Remote_Variables](about_Remote_Variables.md).
 
-To reference .NET Framework namespaces:
+## Syntax
+
+To preload typenames from a .NET namespace:
```
Comment (rjmholt) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md]: I wouldn't say this preloads anything. It just adds prefixes to the type resolution search. See https://github.com/PowerShell/PowerShell/blob/148124e0551df813a5b57f071a5b826d7f44927c/src/System.Management.Automation/engine/parser/TypeResolver.cs#L335-L365.

Maybe something like:

> To specify .NET namespaces from which to resolve types

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md [hunk 3]
```diff
@@ -1,46 +1,63 @@
...
-> module. If the module isn't loaded in the current session, the `using`
-> statement fails.
+To preload classes from a .NET assembly:
+
+```
+using assembly <.NET-assembly-path>
+using assembly <.NET-namespace>
+```
+
+Loading a namespace makes it easier to reference types by their short names.
```
Comment (rjmholt) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md]: This isn't really a form of loading. Just adds places to look for types.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md [hunk 4]
```diff
@@ -1,46 +1,63 @@
...
+Loading an assembly preloads classes into a script at parse time. This allows
```
Comment (rjmholt) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md]:
```suggestion
Loading an assembly preloads .NET types from that assembly into a script at parse time. This allows
```

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md [hunk 5]
```diff
@@ -1,46 +1,63 @@
...
+you to create new PowerShell classes that are derived from the preloaded
```
Comment (rjmholt) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md]:
```suggestion
you to create new PowerShell classes that use types from the preloaded assembly.
```

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md [hunk 6]
```diff
@@ -1,46 +1,63 @@
...
+classes. In Windows PowerShell 5.1 you can load the assembly by path name or by
```
Comment (rjmholt) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md]:
```suggestion
In Windows PowerShell 5.1 you can load the assembly by path name or by
```

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md [hunk 7]
```diff
@@ -1,46 +1,63 @@
...
+namespace. When you use the namespace name, PowerShell searches the .NET Global
```
Comment (rjmholt) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md]:
```suggestion
name. When you use the name, PowerShell searches the .NET Global
```

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md [hunk 8]
```diff
@@ -1,46 +1,63 @@
...
+Assembly Cache (GAC) for the associated DLL.
```
Comment (rjmholt) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md]:
```suggestion
Assembly Cache (GAC) for the associated assembly.
```

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md (hunk 1 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md]: fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md (hunk 2 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md]: Fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md [hunk 9]
```diff
@@ -64,20 +81,58 @@ $hashfromstream = Get-FileHash -InputStream $memorystream `
 $hashfromstream.Hash.ToString()
 ```
 
-The following script assumes a module named 'CardGames' was loaded
-automatically.
+### Example 2 - Load classes from a script module
+
+In this example, we have a PowerShell script module named **CardGames** the
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md]: the last `the` should be `that`.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md (hunk 3 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md]: Fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md (hunk 4 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md]: Fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md [hunk 10]
```diff
@@ -64,20 +81,58 @@ $hashfromstream = Get-FileHash -InputStream $memorystream `
...
-The following classes are defined in the module:
+- **CardGames.Deck**
+- **CardGames.Card**
 
-- **Deck**
-- **Card**
+The `using module` command imports the module. `Import-Module` and the
+`#requires` statement only import the module functions, aliases, and variables,
+as defined by the module. Classes are not imported. The `using module`
+statement imports the classes defined in the module.
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md]: Can reword this? I think just the first sentence could be removed and it'll be clearer.

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md (hunk 6 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md]: Fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md (hunk 5 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md]: Fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md (hunk 7 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md]: Fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md (hunk 8 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md]: Fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md (hunk 9 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md]: Fixed

File: reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md (hunk 10 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Core/About/about_Using.md]: The sentence is needed but it could be clearer.

**Pull request merged**